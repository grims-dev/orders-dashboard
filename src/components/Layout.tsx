import React from "react";
import { ThemeProvider, createGlobalStyle } from "styled-components";
import { Helmet } from "react-helmet";

const theme = {
  red: "#FF0000",
  brandMain: "#CB1E27",
  black: "#393939",
  grey: "#3A3A3A",
  lightgrey: "#E1E1E1",
  offWhite: "#EDEDED",
  maxWidth: "1600px",
  bs: "0 12px 24px 0 rgba(0, 0, 0, 0.09)",
};

const GlobalStyle = createGlobalStyle`
 
  html {
        box-sizing: border-box;
        font-size: 10px;
    }
    *, *:before, *:after {
        box-sizing: inherit;
    }

    :root {
  --status-readytotry: #49D5B0;
  --status-ontheway: #66A8E9;
  --status-inthequeue: #FD6731;
  --status-outofstock: #FB0001;
}

    body {
        margin: 0;
  padding: 0;
  font-family: 'Montserrat', -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  overflow-x: hidden;
  background-color: #222;
  color: #fefefe;
    }
    a {
        text-decoration: none;
        color: ${theme.black};
    }
`;

const Layout = (props: any) => (
  <ThemeProvider theme={theme}>
    <>
      <Helmet>
        <link rel="stylesheet" type="text/css" href="App.css" />
      </Helmet>
      <GlobalStyle />
      {props.children}
    </>
  </ThemeProvider>
);

export default Layout;
