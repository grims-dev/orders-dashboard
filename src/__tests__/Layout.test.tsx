import { mount } from "enzyme";
import * as React from "react";
import toJSON from "enzyme-to-json";
import Layout from "../components/Layout";

describe("Layout renders successfully", () => {
  it("renders without crashing", () => {
    const wrapper = mount(<Layout />);
    const component = wrapper.find('[data-test="layout"]');
    expect(toJSON(component)).toMatchSnapshot();
  });

  it("renders snapshot of Layout", () => {
    const wrapper = mount(<Layout />);
    expect(wrapper).toMatchSnapshot();
  });
});
