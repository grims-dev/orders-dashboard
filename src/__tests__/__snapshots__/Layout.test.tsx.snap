// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders successfully renders snapshot of Layout 1`] = `
<Layout>
  <ThemeProvider
    theme={
      Object {
        "black": "#393939",
        "brandMain": "#CB1E27",
        "bs": "0 12px 24px 0 rgba(0, 0, 0, 0.09)",
        "grey": "#3A3A3A",
        "lightgrey": "#E1E1E1",
        "maxWidth": "1600px",
        "offWhite": "#EDEDED",
        "red": "#FF0000",
      }
    }
  >
    <HelmetWrapper
      defer={true}
      encodeSpecialCharacters={true}
    >
      <SideEffect(NullComponent)
        defer={true}
        encodeSpecialCharacters={true}
        link={
          Array [
            Object {
              "href": "App.css",
              "rel": "stylesheet",
              "type": "text/css",
            },
          ]
        }
      >
        <NullComponent
          defer={true}
          encodeSpecialCharacters={true}
          link={
            Array [
              Object {
                "href": "App.css",
                "rel": "stylesheet",
                "type": "text/css",
              },
            ]
          }
        />
      </SideEffect(NullComponent)>
    </HelmetWrapper>
    <Memo(GlobalStyleComponent) />
  </ThemeProvider>
</Layout>
`;

exports[`Layout renders successfully renders without crashing 1`] = `null`;
